### ISSUE TITLE
Broken on Windows

### ISSUE THREAD

**Issue by gaearon** [waysact/webpack-subresource-integrity#22]
We integrated this plugin into Create React App (https://github.com/facebookincubator/create-react-app/issues/1202), and the build breaks on Windows:

![](https://cloud.githubusercontent.com/assets/5063473/21002858/32a2560c-bd28-11e6-91a4-bda786e7752e.png)

To reproduce:

```
npm i -g create-react-app
create-react-app myapp --scripts-version 0.8.2

cd myapp
npm run build
```

This happens on Windows because the generated compilation assets key contains Windows path separators but Webpack expected a normalized one.

**Comment by jscheid** [comment]
Released in v0.8.1. Sorry about that, we're not using Windows here and it didn't occur to anybody to test with it :-/

I'm planning to setup CI on Windows soon, perhaps with Appveyor.

**Comment by gaearon** [comment]
Sounds great! We don’t test on Windows either but we should :-/

**Comment by jantimon** [comment]
Wow cool now it is part of create react app? :)

**Comment by gaearon** [comment]
Yea!

**Comment by chenEdgar** [comment]
I reproduced this question on window8.1.
I used webpack 1.13.2 and  html-webpack-plugin with injected true.
![qq 20170515200340](https://cloud.githubusercontent.com/assets/12596622/26056904/d195c5a2-39a9-11e7-8d81-5d8c8daf654e.png)

**Comment by jscheid** [comment]
@chenEdgar it does look like the same issue, but that issue was fixed (and we are now running the test suite on Windows) so it's more likely something unrelated. Perhaps it would be better if you opened a new issue.

I'll need more information to see what's going on, did you see this? https://github.com/waysact/webpack-subresource-integrity/blob/master/CONTRIBUTING.md#issues

Does it work with the current stable version of webpack-subresource-integrity (0.9.0)?

Do you have a way to test on a Unix-ey system (Linux or MacOS perhaps) and see if it works there?

Could you add debug output (before this line https://github.com/waysact/webpack-subresource-integrity/blob/6500e5103564f68c32683bcec0f3ac33367dee5f/index.js#L326) so that we can know the values of `src` and of `Object.keys(compilation.assets)`?

**Comment by WallRushGO** [comment]
I had the same problem that was caused by the css file (value: src) missing from the compilation assets (compilation.assets).

What I did was that I changed line 326 to
`return src in compilation.assets ? compilation.assets[src].integrity : false;`

This caused a correct warning in the output:
`WARNING in webpack-subresource-integrity: Cannot determine hash for asset 'css/app.bb3c30f.css', the resource will be unprotected.`

**Comment by WallRushGO** [comment]
This however doesn't explain why in my case it is caused by the css file.

**Comment by WallRushGO** [comment]
Found the reason why our css was crashing:

*Log*
index.js:325 - Assets for type: **css:  [ ....'css\\app.xxxx.css' ...]**
index.js:168 - hwpAssetPath::parameter src= **/css\app.xxxx.css**
index.js:331 - variable filePath = **/css\app.xxxx.css**
index.js:332 - variable src = **css/app.xxxx.css**

When the filePath (/css\app.xxxx.css) enters hwpAssetPath it comes out as css/app.xxxx.css
This however doesn't match the compilation.assets css\\app.xxxx.css

The bug now being that for some reason compilation.assets has a 'css\\<name>' that is wrong (all js files are in the form js/<name>) causing the hwpAssetPath output not to match with the one found in compilation.assets

**Comment by jscheid** [comment]
Thanks for investigating with this, @WallRushGO! I think with this info I'll be able to reproduce it. I'll have a closer look at this over the next few days.

**Comment by WallRushGO** [comment]
The root cause for us was to use the **path.join** in ExtractTextPlugin

```
const extractCSS = new ExtractTextPlugin({
    filename: utils.assetsPath(path.join(config.build.mapping.styles, '[name].[contenthash:7].css'))
})
```

**Our fix**
```
const extractCSS = new ExtractTextPlugin({
    filename: utils.assetsPath([config.build.mapping.styles, '[name].[contenthash:7].css'].join('/'))
})

```

**Comment by jscheid** [comment]
Ah, good to know, but I'd call this a workaround and not a fix. I'm aiming to release a fix for this soon and you'll be able to go back to using `path.join`.

**Comment by WallRushGO** [comment]
👍

**Comment by bzbetty** [comment]
I'm using ExtractTextPlugin("styles.css?[contenthash]");

so for me it appears to be searching for

styles.css?b201f92da5f06ce3a6eb8816602e1230&cd9dfb6f4017ef825e33

but the compilation.assets dictionary only contains an entry for

styles.css?b201f92da5f06ce3a6eb8816602e1230

unsure if this is totally related to this particular issue, but I am running this on windows and it also seems to be css + extract text related as @WallRushGO 's problem was.

**Comment by jscheid** [comment]
@WallRushGO @bzbetty both should be fixed in v1.0.3, let me know how it works for you.

**Comment by bzbetty** [comment]
yep works great now.
